File: programmers/문자열압축.cpp
#include <iostream>
#include <string>

using namespace std;

int solution(string s){
    int answer = 0;
    int min_length = 987654321;
    
    for (int i=1;i<=s.size()/2+1;i++){
        string prev = s.substr(0,i);
        int cnt = 1;
        string change_s = prev;
        for (int j=i;j<s.size();j+=i){
            string now = s.substr(j,i);
            if (prev == now){
                cnt++;
            } else if (cnt == 1){
                change_s += now;
            }
            else {
                change_s += to_string(cnt);
                change_s += now;
                cnt = 1;
            }
            prev = now;
        }
        if (cnt != 1){
            change_s += to_string(cnt);
        }

        if (min_length>change_s.size()){
            min_length = change_s.size();
        }
    }
    return min_length;
}

int main(){

    cout << solution("a") << "\n";
    return 0;
}

File: programmers/level2/문자열압축.py
def solution(s):
    if len(s) == 1:
        return 1
    min_length = 987654321

    for i in range(1, len(s) // 2 + 1):
        cnt = 1
        now_len = 0
        prev = s[0:i]

        for idx in range(i, len(s), i):
            now = s[idx : idx + i]

            if now == prev:
                cnt += 1

            else:
                now_len += len(prev)
                if cnt != 1:
                    now_len += len(str(cnt))
                cnt = 1
            prev = now
        now_len += len(prev)
        if cnt != 1:
            now_len += len(str(cnt))

        min_length = min(min_length, now_len)

    return min_length


# aabbaccc
